cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/etl/realtime/ETLWorker.java

```diff
@@ -122,13 +132,24 @@ public void configure() {
       setResources(config.getResources());
     }
 
-    PipelineRegisterer registerer = new PipelineRegisterer(getConfigurer(), "realtime");
-    // using table dataset type for error dataset
-    PipelinePhase pipeline = registerer.registerPlugins(config, Table.class, DatasetProperties.builder()
-      .add(Table.PROPERTY_SCHEMA, ERROR_SCHEMA.toString())
-      .build(), false);
+    PipelineSpecGenerator specGenerator =
+      new PipelineSpecGenerator(getConfigurer(),
+                                RealtimeSource.PLUGIN_TYPE,
+                                RealtimeSink.PLUGIN_TYPE,
+                                Table.class,
+                                DatasetProperties.builder()
+                                  .add(Table.PROPERTY_SCHEMA, ERROR_SCHEMA.toString())
+                                  .build());
+    PipelineSpec spec = specGenerator.generateSpec(config);
+    PipelinePlanner planner = new PipelinePlanner(SUPPORTED_PLUGIN_TYPES, ImmutableSet.<String>of());
+    PipelinePlan plan = planner.plan(spec);
+    if (plan.getPhases().size() > 1) {
+      throw new IllegalArgumentException("Only one source is supported at this time.");
```

Comment:
Do you mean `Only one phase is supported at this time.`?
Or is there the correlation defined that the number of phases corresponds to the number of sinks?